shared/p2p/service.go

```diff
@@ -57,30 +50,23 @@ func NewServer() (*Server, error) {
 	}
 
 	return &Server{
-		ctx:    ctx,
-		cancel: cancel,
-		feeds:  make(map[reflect.Type]*event.Feed),
-		host:   host,
-		gsub:   gsub,
-		mutex:  &sync.Mutex{},
+		ctx:          ctx,
+		cancel:       cancel,
+		feeds:        make(map[reflect.Type]*event.Feed),
+		host:         host,
+		gsub:         gsub,
+		mutex:        &sync.Mutex{},
+		topicMapping: make(map[reflect.Type]string),
 	}, nil
 }
 
 // Start the main routine for an p2p server.
 func (s *Server) Start() {
 	log.Info("Starting service")
-	if err := startDiscovery(s.ctx, s.host, s.gsub); err != nil {
+	if err := startDiscovery(s.ctx, s.host); err != nil {
 		log.Errorf("Could not start p2p discovery! %v", err)
 		return
 	}
-
-	// Subscribe to all topics.
-	for topic, msgType := range topicTypeMapping {
-		log.WithFields(logrus.Fields{
-			"topic": topic,
-		}).Debug("Subscribing to topic")
-		go s.subscribeToTopic(topic, msgType)
```

Comment:
Should we remove this method, `subscribeToTopic`?

Reply:
I am surprised this didn't upset the `unused` linter. It was only used in tests!